compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/types/Types.java

```diff
@@ -0,0 +1,201 @@
+    public final TypeSymbol ANYDATA;
+    public final TypeSymbol NEVER;
+    public final TypeSymbol READONLY;
+    public final TypeSymbol JSON;
+    public final TypeSymbol BYTE;
+    public final TypeSymbol COMPILATION_ERROR;
+    public final CompilerContext context;
+
+
+    private Types(CompilerContext context) {
+        context.put(TYPES_KEY, this);
+        this.context = context;
+        TypesFactory typesFactory = TypesFactory.getInstance(context);
+        this.symbolFactory = SymbolFactory.getInstance(context);
+        this.symbolTable = SymbolTable.getInstance(context);
+        this.packageCache = PackageCache.getInstance(context);
+
+        this.BOOLEAN = typesFactory.getTypeDescriptor(symbolTable.booleanType);
+        this.INT = typesFactory.getTypeDescriptor(symbolTable.intType);
+        this.FLOAT = typesFactory.getTypeDescriptor(symbolTable.floatType);
+        this.DECIMAL = typesFactory.getTypeDescriptor(symbolTable.decimalType);
+        this.STRING = typesFactory.getTypeDescriptor(symbolTable.stringType);
+        this.NIL = typesFactory.getTypeDescriptor(symbolTable.nilType);
+        this.XML = typesFactory.getTypeDescriptor(symbolTable.xmlType);
+        this.ERROR = typesFactory.getTypeDescriptor(symbolTable.errorType);
+        this.FUNCTION = typesFactory.getTypeDescriptor(symbolTable.invokableType);
+        this.FUTURE = typesFactory.getTypeDescriptor(symbolTable.futureType);
+        this.TYPEDESC = typesFactory.getTypeDescriptor(symbolTable.typeDesc);
+        this.HANDLE = typesFactory.getTypeDescriptor(symbolTable.handleType);
+        this.STREAM = typesFactory.getTypeDescriptor(symbolTable.streamType);
+        this.ANY = typesFactory.getTypeDescriptor(symbolTable.anyType);
+        this.ANYDATA = typesFactory.getTypeDescriptor(symbolTable.anydataType);
+        this.NEVER = typesFactory.getTypeDescriptor(symbolTable.neverType);
+        this.READONLY = typesFactory.getTypeDescriptor(symbolTable.readonlyType);
+        this.JSON = typesFactory.getTypeDescriptor(symbolTable.jsonType);
+        this.BYTE = typesFactory.getTypeDescriptor(symbolTable.byteType);
+        this.COMPILATION_ERROR = typesFactory.getTypeDescriptor(symbolTable.semanticError);
+    }
+
+    /**
+     * Lookup for the symbol of a user defined type within a given module. This would be considering type
+     * definitions, enums, enum members, and class definitions as valid user defined types when
+     * looking up. The module is determined by the provided org, module name and the version.
+     *
+     * @param org           The organization of the looking up module
+     * @param moduleName    The name of the looking up module
+     * @param version       The version of the looking up module
+     * @param typeDefName   The type definition
+     * @return The {@link Symbol} of the user defined type
+     */
+    public Optional<Symbol> getTypeByName(String org, String moduleName, String version, String typeDefName) {
```

Comment:
The doc comments for an API needs to be comprehensive and makes up part of the API. It needs to define the behaviour of the API which cannot be conveyed through the method signature. For example, there's nothing here about the conditions in which this would return `Optional.empty()`.

Reply:
This applies to all the methods that make up the API